scripts/core/editor3/components/spellchecker/SpellcheckerContextMenu.jsx

```diff
@@ -106,9 +105,7 @@ SpellcheckerContextMenuComponent.propTypes = {
  * @description Maps the values from state to the component callback type props.
  */
 const mapDispatchToProps = (dispatch) => ({
-    addWord: (word) => dispatch(actions.ignoreWord(word)),
-    ignoreWord: (word) => dispatch(actions.addWord(word)),
-    replaceWord: (word) => setTimeout(() => dispatch(actions.replaceWord(word)), 0),
+    replaceWord: (word) => setTimeout(() => dispatch(actions.replaceWord(word)), 0)
```

Comment:
When the context menu is closed the editor automatically updates his status. Also on  word replace action the editor is updated again. If those actions are too closed, the replace will have no results. So there is this setTimeout call with 0s that solves the issue. Here are more info about setTimeout trick:
https://www.quora.com/What-does-setTimeout-with-a-0ms-delay-do